drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/PackageModel.java

```diff
@@ -112,47 +113,49 @@
     
-    private Set<String> imports = new HashSet<>();
-    private Set<String> staticImports = new HashSet<>();
-    private Set<String> entryPoints = new HashSet<>();
+    private final Set<String> imports = new HashSet<>();
+    private final Set<String> staticImports = new HashSet<>();
+    private final Set<String> entryPoints = new HashSet<>();
     private Map<String, Method> staticMethods;
 
-    private Map<String, Class<?>> globals = new HashMap<>();
+    private final Map<String, Class<?>> globals = new HashMap<>();
 
-    private Map<String, Map<Integer, MethodDeclaration>> ruleMethods = new ConcurrentHashMap<>();
+    private final Map<String, Map<Integer, MethodDeclaration>> ruleMethods = new ConcurrentHashMap<>();
 
-    private Set<String> queryNames = new HashSet<>();
-    private Map<String, MethodDeclaration> queryMethods = new ConcurrentHashMap<>();
-    private Map<String, Set<QueryModel>> queriesByRuleUnit = new ConcurrentHashMap<>();
+    private final Set<String> queryNames = new HashSet<>();
+    private final Map<String, MethodDeclaration> queryMethods = new ConcurrentHashMap<>();
+    private final Map<String, Set<QueryModel>> queriesByRuleUnit = new ConcurrentHashMap<>();
 
-    private Map<String, QueryGenerator.QueryDefWithType> queryDefWithType = new HashMap<>();
+    private final Map<String, QueryGenerator.QueryDefWithType> queryDefWithType = new HashMap<>();
 
-    private Map<String, MethodCallExpr> windowReferences = new HashMap<>();
+    private final Map<String, MethodCallExpr> windowReferences = new HashMap<>();
 
-    private Map<String, List<QueryParameter>> queryVariables = new HashMap<>();
+    private final Map<String, List<QueryParameter>> queryVariables = new HashMap<>();
 
-    private List<MethodDeclaration> functions = new ArrayList<>();
+    private final List<MethodDeclaration> functions = new ArrayList<>();
 
-    private List<TypeDeclaration> generatedPOJOs = new ArrayList<>();
-    private List<GeneratedClassWithPackage> generatedAccumulateClasses = new ArrayList<>();
+    private final List<TypeDeclaration> generatedPOJOs = new ArrayList<>();
+    private final List<GeneratedClassWithPackage> generatedAccumulateClasses = new ArrayList<>();
 
-    private Set<Class<?>> domainClasses = new HashSet<>();
-    private Map<Class<?>, ClassDefinition> classDefinitionsMap = new HashMap<>();
+    private final Set<Class<?>> domainClasses = new HashSet<>();
+    private final Map<Class<?>, ClassDefinition> classDefinitionsMap = new HashMap<>();
 
-    private List<Expression> typeMetaDataExpressions = new ArrayList<>();
+    private final List<Expression> typeMetaDataExpressions = new ArrayList<>();
 
-    private DRLIdGenerator exprIdGenerator;
+    private final DRLIdGenerator exprIdGenerator;
 
-    private KnowledgeBuilderConfigurationImpl configuration;
+    private final KnowledgeBuilderConfigurationImpl configuration;
     private Map<String, AccumulateFunction> accumulateFunctions;
```

Comment:
What about
```suggestion
    private final Map<String, AccumulateFunction> accumulateFunctions = new HashMap();
```

And then https://github.com/kiegroup/drools/blob/cd1c2e5a6b32a8494d7c6dc73e94e78161ce79ee/drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/PackageModel.java#L820
just perform a 
`accumilateFunctions.putAll` 
instead of just take the instance from outside.
It should be safer

Reply:
If I do a `putAll` I'll unnecessarily move the map's entries from a map to another instead of using it as it is.